plugins/modules/mso_rest.py

```diff
@@ -0,0 +1,208 @@
+extends_documentation_fragment:
+- cisco.mso.modules
+
+notes:
+- Most payloads are known not to be idempotent, so be careful when constructing payloads.
+seealso:
+- module: cisco.mso.mso_tenant
+author:
+- Anvitha Jain (@anvitha-jain)
+'''
+
+EXAMPLES = r'''
+- name: Add schema (JSON)
+  cisco.mso.mso_rest:
+    host: mso
+    username: admin
+    password: SomeSecretPassword
+    path: /mso/api/v1/schemas
+    method: post
+    content:
+      {
+          "displayName": "mso_schema",
+          "templates": [{
+              "name": "Template_1",
+              "tenantId": "{{ add_tenant.jsondata.id }}",
+              "displayName": "Template_1",
+              "templateSubType": [],
+              "templateType": "stretched-template",
+              "anps": [],
+              "contracts": [],
+              "vrfs": [],
+              "bds": [],
+              "filters": [],
+              "externalEpgs": [],
+              "serviceGraphs": [],
+              "intersiteL3outs": []
+          }],
+          "sites": [],
+          "_updateVersion": 0
+      }
+  delegate_to: localhost
+
+- name: Query schema
+  cisco.mso.mso_rest:
+    host: mso
+    username: admin
+    password: SomeSecretPassword
+    path: /mso/api/v1/schemas
+    method: get
+  delegate_to: localhost
+
+- name: Patch schema (YAML)
+  cisco.mso.mso_rest:
+    host: mso
+    username: admin
+    password: SomeSecretPassword
+    path: "/mso/api/v1/schemas/{{ add_schema.jsondata.id }}"
+    method: patch
+    content:
+      - op: add
+        path: /templates/Template_1/anps/-
+        value:
+          name: AP2
+          displayName: AP2
+          epgs: []
+        _updateVersion: 0
+  delegate_to: localhost
+'''
+
+RETURN = r'''
+'''
+
+import json
+import os
+
+# Optional, only used for YAML validation
+try:
+    import yaml
+    HAS_YAML = True
+except Exception:
+    HAS_YAML = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import MSOModule, mso_argument_spec
+from ansible.module_utils.urls import fetch_url
+from ansible.module_utils._text import to_text
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        path=dict(type='str', required=True, aliases=['uri']),
+        method=dict(type='str', default='get', choices=['delete', 'get', 'post', 'put', 'patch'], aliases=['action']),
+        content=dict(type='raw', aliases=['payload']),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+    )
+
+    content = module.params.get('content')
+    path = module.params.get('path')
+
+    mso = MSOModule(module)
+    mso.result['status'] = -1  # Ensure we always return a status
+
+    # Validate content/payload
+    if content and (isinstance(content, dict) or isinstance(content, list)):
+        # Validate inline YAML/JSON
+        content = json.dumps(content)
+    elif content and isinstance(content, str) and HAS_YAML:
```

Comment:
What happens when content is a string and HAS_YAML is false?

Reply:
If HAS_YAML is false then it breaks.